ctags.go

```diff
@@ -222,37 +217,27 @@ func (p *ctagsProcess) Parse(name string, content []byte) ([]*Entry, error) {
 	}
 
 	if !utf8.Valid(content) {
-		if p.Info != nil {
-			p.Info.Printf("ctags skipping file due not being utf-8 encoded: %s", name)
-		}
-		return nil, nil
+		return nil, NewError("file is not utf-8 encoded", name)
 	}
 
 	if ok, err := p.post(&req, content); err != nil {
-		return nil, err
+		return nil, NewFatalError(err.Error(), name)
 	} else if !ok {
-		if p.Info != nil {
-			p.Info.Printf("ctags skipping file due to long filename: %s", name)
-		}
-		return nil, nil
+		return nil, NewError("filename is too long", name)
 	}
 
 	// 250 is a better guess for initial size
 	es := make([]*Entry, 0, 250)
 	for {
 		var rep reply
 		if err := p.read(&rep); err != nil {
-			return nil, err
+			return nil, NewFatalError(rep.Message, name)
```

Comment:
Isn't this wrong? Shouldn't you be returning err here. I assume when err is non-nil rep will be empty.

Reply:
Oops, sorry for the sloppy PR here. I was apparently very sleepy when opening this.